_publications/2022-07-26-portmaster-vs-glasswire.md

```diff
@@ -0,0 +1,208 @@
+
+You can even block risky connection types system-wide, such as p2p or incoming, and then create exceptions for trusted apps.
+
+### GlassWire Seems to Be A Proxy for the Windows Filtering Platform
+
+Disclaimer: GlassWire is closed-source - therefore we had to analyze their software in a blackbox test. We do not have the full picture on how GlassWire integrates into the operating system.
+
+What we did find out is that GlassWire uses the Windows Filtering Platform (WFP) in user-space mode, creating rules for applications essentially exactly like the Windows Firewall does itself. This is a similar approach to that of [Simplewall]({{ site.simplewall_comparison_url }}). When Windows Firewall is disabled, this also makes GlassWire blind and incapable of blocking anything.
+
+GlassWire also uses a background service to collect additional data about connections. It is hard to find out what this service exactly does.
+
+Finally, an oddity we came across was that GlassWire provides a tray icon with the "Exit" option. But pressing that does not end GlassWire or the service as expected - instead the software continues to do what it does. We feel that this is unfavorable wording and might not align with user expectation.
+
+### Portmaster Integrates Into the System Using the Kernel Extension
+
+On the other hand, Portmaster uses the Windows Filtering Platform (WFP) in kernel-space mode. This means that it has its own kernel extension with custom logic. This kernel extension sees every network packet that goes in or out of the device. The Portmaster can analyze the packet data to extract information from it, such as domain names and encryption settings of HTTPS connections. As a result it can make a lot of smart decisions.
+
+In order to give you an overview of what is happening on your device, the Portmaster directly feeds the raw network data into its Network monitor, showing you what your network has been up to within the last 10 minutes.
+
+Contrary to GlassWire, Portmaster's system integration is both [documented](https://docs.safing.io/portmaster/architecture/os-integration#windows) and [open source](https://github.com/safing/portmaster-windows-kext/), so you can verify our claims.
+
+### Portmaster Safeguards Your DNS queries From ISPs to Enhance Your Privacy
+
+Encrypted DNS queries allow you to bypass Internet Service Providers (ISPs) and eavesdroppers and directly resolve your requests at the DNS provider. While the ISPs or middleware can no longer easily determine the websites you browse or the apps you use, but your DNS provider definitely can. That being said, even if your ISP is uninformed of the websites you visit, they can monitor the IP addresses you connect to. Since many services employ dedicated static IPs for their infrastructure, ISPs can try to track your queries using conditional logic. So even though DNS encryption improves your privacy, it cannot safeguard all your connections.
```

Comment:
This just looks at the privacy aspect of encrypted DNS, but there are important security (and therefore, also privacy) benefits.
Also, 3/4 of this paragraph is disclaiming what secure DNS cannot do. This seems odd when reading.